src/modules/Dropdown/Dropdown.js

```diff
@@ -686,10 +702,12 @@ export default class Dropdown extends Component {
       search && searchQuery && 'filtered'
     )
     let _text = placeholder
-    if (text && !searchQuery) {
-      _text = text
-    } else if (searchQuery) {
+    if (searchQuery) {
       _text = null
+    } else if (text) {
+      _text = text
+    } else if (open && !multiple) {
+      _text = this.getSelectedItemText()
```

Comment:
@levithomason The two lines are not exactly the same. Line 710 gets the text of the currently highlighted item (by going up and down with the arrow keys), while line 712 gets the current selected item (the `value` in state). In hindsight maybe `getHighlightedItemText()` might be more indicative?

Line 710 adds the behavior that the 'placeholder' value changes with what the value would become when you would blur the dropdown (as it is with the vanilla SUI component).